### ISSUE TITLE
Hebrew Fonts

### ISSUE THREAD

**Issue by adishavit** [nodebox/opentype.js#119]
I noticed that when loading a right-to-left font (like a Hebrew font), the letters get generated from left to right.  
Is this by design?  
If I reverse the input string (assuming I know the language and there are no mixed characters), would the kerning be correct?

**Comment by fdb** [comment]
You're right, we don't take direction into account when rendering the font, assuming LTR direction.

I wouldn't trust kerning values in the other direction. You should reverse kerning pairs then as well.

Is there an open-source Hebrew font I can test with?

**Comment by davelab6** [comment]
http://fonts.gstatic.com/ea/alefhebrew/v4/download.zip

http://fonts.gstatic.com/ea/opensanshebrew/v3/download.zip

http://fonts.gstatic.com/ea/opensanshebrewcondensed/v3/download.zip

http://fonts.gstatic.com/ea/notosanshebrew/v3/download.zip

**Comment by adishavit** [comment]
[This site](http://www.oketz.com/introduction/) has lots of info about Hebrew fonts and [a bunch of them for free](http://www.oketz.com/fonts/good.html).  

The [Ubuntu font](http://font.ubuntu.com/) also supports Hebrew.

**Comment by adishavit** [comment]
Another issue with blind reversal might be diacritics: לַגּוֹפָן יֵשׁ נִקּוּד מְתֻכְנָת
Any string reveral will have to make sure these are resevred as well, so that: אָלֶף is not turned [ףֶלָא](https://mothereff.in/reverse-string#%D7%90%D6%B8%D7%9C%D6%B6%D7%A3) (and this is with a tool designed for reversing Unicode strings!)

**Comment by davelab6** [comment]
On 21 May 2015 at 15:36, adishavit <notifications@github.com> wrote:

> The Ubuntu font <http://font.ubuntu.com/> also supports Hebrew.
>
It was designed but not yet published :( cc @sladen

**Comment by adishavit** [comment]
@davelab6 : Yeah, that's why I removed it...

**Comment by davelab6** [comment]
Sorry, replying via email :) I'll edit the web version now

**Comment by adishavit** [comment]
Just found this discussion [here](https://github.com/mathiasbynens/esrever/issues/5).

**Comment by sladen** [comment]
Indeed, and even more one design was done for Hebrew.  The addition of the Arabic/Hebrew (both with full diacriticals) to existing base .ttf caused regressions by changing the line-height metrics.  There is something in the works for solving that.  If somebody needs an alpha version for eg. doing some work with diacriticals on the two Semitic scripts I'm happy to assist in getting one; but it's not ready for distribution/betaing until a solution is found that doesn't reflow the line-height metrics in documents and the UI for existing user-cases.  See animation in comment #1 of

  "[nc] Line spacing causes major UI problems"
  https://bugs.launchpad.net/ubuntu-font-family/+bug/1394914

**Comment by fdb** [comment]
@davelab6 I'm curious – what's the fonts.gstatic.com domain? Is that an "official" place to download OTF fonts?

**Comment by davelab6** [comment]
On 21 May 2015 at 17:18, Frederik De Bleser <notifications@github.com>
wrote:

> @davelab6 <https://github.com/davelab6> I'm curious – what's the
> fonts.gstatic.com domain? Is that an "official" place to download OTF
> fonts?
>
Its a Google CDN domain. The links are from www.google.com/fonts/earlyaccess

**Comment by fdb** [comment]
Thanks — I wondered what the `/ea/` was for.

**Comment by davelab6** [comment]
On 21 May 2015 at 17:05, Paul Sladen <notifications@github.com> wrote:

> Indeed, and even more one design was done for Hebrew. The addition of the
> Arabic/Hebrew (both with full diacriticals) to existing base .ttf caused
> regressions by changing the line-height metrics. There is something in the
> works for solving that. If somebody needs an alpha version for eg. doing
> some work with diacriticals on the two Semitic scripts I'm happy to assist
> in getting one; but it's not ready for distribution/betaing until a
> solution is found that doesn't reflow the line-height metrics in documents
> and the UI for existing user-cases. See animation in comment #1
> <https://github.com/nodebox/opentype.js/pull/1> of
>
>
The problem is that if you do this, the new scripts will be VERY small at
the same pt/px size... its better to release them as separate families, and
use fontconfig rules to fall back from one to the other.

**Comment by forresto** [comment]
Has anybody looked into this? Arabic support seems good, which also has RTL scripts.

Here is my test project. ([source](https://cuttle.xyz/@forresto/RTL-scripts-text-tests-wJh7xyMBYSTQ))

> ![image](https://github.com/opentypejs/opentype.js/assets/395307/0dc7d253-2b90-4fdf-abb6-e8dcd94789a6)

Hebrew is puts the characters in the wrong order. In addition, the marks are misplaced in the basic example.
